packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -1128,6 +1149,32 @@ function ChildReconciler(shouldTrackSideEffects) {
       child = child.sibling;
```

Comment:
The element.type won't line up on line 1125 which means that it'll kill the state of the subtree every time this rerenders.